pkg/volume/csi/csi_mounter.go

```diff
@@ -164,10 +229,28 @@ func (c *csiMountMgr) TearDown() error {
 func (c *csiMountMgr) TearDownAt(dir string) error {
 	glog.V(4).Infof(log("Unmounter.TearDown(%s)", dir))
 
-	// extract driverName and volID from path
-	base, volID := path.Split(dir)
-	volID = kstrings.UnescapeQualifiedNameForDisk(volID)
-	driverName := path.Base(base)
+	// is dir even mounted ?
+	mounted, err := isDirMounted(c.plugin, dir)
+	if err != nil {
+		glog.Error(log("unmounter.Teardown failed while checking mount status for dir [%s]: %v", dir, err))
+		return err
+	}
+
+	if !mounted {
+		glog.V(4).Info(log("unmounter.Teardown skipping unmout, dir not mounted [%s]", dir))
```

Comment:
We maybe, just maybe, should not have this check. Consider if a SP backs a storage volume using the local root FS (for something like an `emptyDir` or `localStorage`. We can leave it as is, and add a TODO to revisit in the future.

Reply:
+1 should revert this check.

Reply:
Agreed. Please open an issue to track this and create a PR to remove this check. We must leave it up to the CSI driver to decide what tear down means.

CC @vladimirvivien @msau42

Reply:
PR #72008 is addressing this